server/proxy/grpcproxy/watch.go

```diff
@@ -271,6 +272,14 @@ func (wps *watchProxyStream) recvLoop() error {
 		case *pb.WatchRequest_CancelRequest:
 			wps.delete(uv.CancelRequest.WatchId)
 			wps.lg.Debug("cancel watcher", zap.Int64("watcherId", uv.CancelRequest.WatchId))
+		case *pb.WatchRequest_ProgressRequest:
+			wps.mu.Lock()
+			for _, wps := range wps.watchers {
+				wps.requestedProgress = true
+			}
+			wps.mu.Unlock()
+			wps.ranges.wp.cw.RequestProgress(wps.ranges.wp.ctx)
+			wps.lg.Debug("request watcher progress")
```

Comment:
This seems like a new feature. I would suggest to address in a separate discussion session.

Reply:
Yes, I see your point. I'll create a new PR.

Reply:
Done: https://github.com/etcd-io/etcd/pull/15339

Reply:
Please remove all grpcproxy related change from this PR.

Reply:
The test case won't pass then - we either need to merge https://github.com/etcd-io/etcd/pull/15339 first or disable the test case for grpcproxy here. Not sure what'd be preferred?

Reply:
Definitely the latter for now: `disable the test case for grpcproxy here.`. We need to backport this PR to 3.5 and 3.4, and usually we don't backport new features.